Code/Framework/AzFramework/AzFramework/PaintBrush/PaintBrushSettings.cpp

```diff
@@ -58,51 +57,51 @@ namespace AzFramework
 
     bool PaintBrushSettings::GetSizeVisibility() const
     {
-        return (m_brushMode != PaintBrushMode::Eyedropper);
+        return true;
     }
```

Comment:
(repeated) since all of these visibility settings seem to be conditional, should they all default to `false`? Or some of them?

Reply:
No, they actually very much _need_ to default to true in the PaintBrushSettings, and then use the conditional logic in the overrides in GlobalPaintBrushSettings.

The issue that this fixes is that the GlobalPaintBrushSettings, which are the ones used in the Editor from the paintbrush tool, show and hide various fields based on whether you're using the paintbrush, eyedropper, or smoothing tool. However, if you try to use PaintBrushSettings from ScriptCanvas, the SC API for painting exposes those tools as separate APIs, not separate modes, so you need to be able to edit _all_ the settings for the PaintBrushSettings if you create an SC variable containing one of these. What was happening is that with the conditionals on here, you could never see or edit the color field or the smoothing values from SC, because it would only show them if the brush mode changed which literally couldn't happen in the SC use case.

Reply:
> No, they actually very much _need_ to default to true in the PaintBrushSettings, and then use the conditional logic in the overrides in GlobalPaintBrushSettings.
> 
> The issue that this fixes is that the GlobalPaintBrushSettings, which are the ones used in the Editor from the paintbrush tool, show and hide various fields based on whether you're using the paintbrush, eyedropper, or smoothing tool. However, if you try to use PaintBrushSettings from ScriptCanvas, the SC API for painting exposes those tools as separate APIs, not separate modes, so you need to be able to edit _all_ the settings for the PaintBrushSettings if you create an SC variable containing one of these. What was happening is that with the conditionals on here, you could never see or edit the color field or the smoothing values from SC, because it would only show them if the brush mode changed which literally couldn't happen in the SC use case.

Oh the visibility actually prevents the fields from being edited programmatically? :/ Not sure that makes a lot of sense, but I understand now.

Reply:
It doesn't prevent it from being edited programmatically, but if you create an SC variable and set it to "initialize from component", the edit context for the structure gets displayed on the component card in the entity inspector. _That's_ the point at which the visibility affects what you can and can't initialize.

Reply:
![image](https://user-images.githubusercontent.com/82224783/209986768-cd38ca44-1e2c-4187-b9f1-e4810c1b0f37.png)

The circled fields weren't showing up before, and the Brush Mode *was* showing up even though it's meaningless in this context.